storage/database/db_manager_test.go

```diff
@@ -672,24 +673,212 @@ func TestDBManager_Governance(t *testing.T) {
 	// TODO-Klaytn-Database Implement this!
 }
 
-func TestDBManager_StateTrieMigration(t *testing.T) {
+func TestDatabaseManager_CreateMigrationDBAndSetStatus(t *testing.T) {
+	for i, dbm := range dbManagers {
+		if dbConfigs[i].DBType == MemoryDB {
+			continue
+		}
+
+		// check if migration fails on single DB
+		if !dbm.IsPartitioned() {
+			migrationBlockNum := uint64(12345)
+
+			// check if not in migration
+			assert.False(t, dbManagers[i].InMigration())
+
+			// check if create migration fails
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
+			assert.Error(t, err) // expect error
+
+			continue
+		}
+
+		// check if migration fails when in migration
+		{
+			migrationBlockNum := uint64(34567)
+
+			// check if not in migration
+			assert.False(t, dbManagers[i].InMigration())
+
+			// set migration status
+			dbm.setStateTrieMigrationStatus(migrationBlockNum)
+			assert.True(t, dbManagers[i].InMigration())
+
+			// check if create migration fails
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
+			assert.Error(t, err) // expect error
+
+			// reset migration status for next test
+			dbm.setStateTrieMigrationStatus(0)
+		}
+
+		// check if CreateMigrationDBAndSetStatus works as expected
+		{
+			migrationBlockNum := uint64(56789)
+
+			// check if not in migration state
+			assert.False(t, dbManagers[i].InMigration())
+
+			err := dbm.CreateMigrationDBAndSetStatus(migrationBlockNum)
```

Comment:
Don't we need to remove created migrationDB?

Reply:
Thanks :) I put FinishStateMigration after the test finishes